tests/contrib/test_sqlalchemy.py

```diff
@@ -0,0 +1,120 @@
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
+
+from oauth2client import GOOGLE_TOKEN_URI
+from oauth2client.client import OAuth2Credentials
+from oauth2client.contrib.sqlalchemy import CredentialsType, Storage
+
+Base = declarative_base()
+
+
+class DummyModel(Base):
+    __tablename__ = 'dummy'
+
+    id = Column(Integer, primary_key=True)
+    key = Column(Integer)  # we will query against this, because of ROWID
+    credentials = Column(CredentialsType)
+
+
+class TestSQLAlchemyStorage(unittest.TestCase):
+    engine = create_engine('sqlite://')
+
+    @classmethod
+    def setUpClass(cls):
+        Base.metadata.create_all(cls.engine)
+        cls.session = sessionmaker(bind=cls.engine)
+
+    def setUp(self):
+        self.credentials = OAuth2Credentials(
+            access_token='token',
+            client_id='client_id',
+            client_secret='client_secret',
+            refresh_token='refresh_token',
+            token_expiry=datetime.datetime.utcnow(),
+            token_uri=GOOGLE_TOKEN_URI,
+            user_agent='DummyAgent',
+        )
+
+    def tearDown(self):
+        session = self.session()
+        session.query(DummyModel).filter_by(key=1).delete()
+        session.commit()
+
+    def compare_credentials(self, result):
+        self.assertEqual(result.access_token, self.credentials.access_token)
+        self.assertEqual(result.client_id, self.credentials.client_id)
+        self.assertEqual(result.client_secret, self.credentials.client_secret)
+        self.assertEqual(result.refresh_token, self.credentials.refresh_token)
+        self.assertEqual(result.token_expiry, self.credentials.token_expiry)
+        self.assertEqual(result.token_uri, self.credentials.token_uri)
+        self.assertEqual(result.user_agent, self.credentials.user_agent)
+
+    def test_get(self):
+        session = self.session()
+        session.add(DummyModel(
+            key=1,
+            credentials=self.credentials,
+        ))
+        session.commit()
+
+        ret = Storage(
+            session=session,
+            model_class=DummyModel,
+            key_name='key',
+            key_value=1,
+            property_name='credentials',
+        ).get()
+
+        self.compare_credentials(ret)
+
+    def test_put(self):
+        session = self.session()
+        Storage(
+            session=session,
+            model_class=DummyModel,
+            key_name='key',
+            key_value=1,
+            property_name='credentials',
+        ).put(self.credentials)
+        session.commit()
+
+        ret = session.query(DummyModel).filter_by(key=1).first()
+        self.compare_credentials(ret.credentials)
+
+    def test_delete(self):
+        session = self.session()
+        session.add(DummyModel(
+            key=1,
+            credentials=self.credentials,
+        ))
+        session.commit()
+
+        q = session.query(DummyModel).filter_by(key=1)
```

Comment:
[Single-letter identifiers are forbidden by the style guide](https://google.github.io/styleguide/pyguide.html?showone=Naming#Naming).